### PR TITLE
Add default mode support

### PR DESCRIPTION
**Pull request by anuruddhal**
## Purpose
- Add default mode support
- Fix https://github.com/ballerinax/kubernetes/issues/495

### CODE REVIEW

File: kubernetes-extension/src/main/java/org/ballerinax/kubernetes/handlers/DeploymentHandler.java [hunk 1]
```diff
@@ -185,6 +185,9 @@ private Container generateContainer(DeploymentModel deploymentModel, List<Contai
                     .withSecretName(secretModel.getName())
                     .endSecret()
                     .build();
+            if (secretModel.getDefaultMode() > 0) {
```
Comment (hemikak) [inline kubernetes-extension/src/main/java/org/ballerinax/kubernetes/handlers/DeploymentHandler.java]: Shall we keep an empty line above ?

File: kubernetes-extension/src/main/java/org/ballerinax/kubernetes/handlers/DeploymentHandler.java [hunk 2]
```diff
@@ -194,6 +197,9 @@ private Container generateContainer(DeploymentModel deploymentModel, List<Contai
                     .withName(configMapModel.getName())
                     .endConfigMap()
                     .build();
+            if (configMapModel.getDefaultMode() > 0) {
```
Comment (hemikak) [inline kubernetes-extension/src/main/java/org/ballerinax/kubernetes/handlers/DeploymentHandler.java]: Shall we keep an empty line above ?

File: kubernetes-extension-test/src/test/resources/deployment/default_mode.bal [hunk 3]
```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (hemikak) [inline kubernetes-extension-test/src/test/resources/deployment/default_mode.bal]: Need to update the year

File: kubernetes-extension-test/src/test/resources/default-mode/default_mode_secret.bal [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import ballerina/http;
```
Comment (hemikak) [inline kubernetes-extension-test/src/test/resources/default-mode/default_mode_secret.bal]: Need to add the copyright header.

File: kubernetes-extension-test/src/test/resources/default-mode/default_mode_config_map.bal [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import ballerina/http;
```
Comment (hemikak) [inline kubernetes-extension-test/src/test/resources/default-mode/default_mode_config_map.bal]: Need to add the copyright header

File: kubernetes-extension-test/src/test/resources/deployment/default_mode.bal (hunk 3 above)
Comment (hemikak) [inline kubernetes-extension-test/src/test/resources/deployment/default_mode.bal]: Do we need this file ? its not being used.

File: kubernetes-extension-test/src/test/resources/default-mode/default_mode_config_map.bal [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    singleYAML: false
+}
+@kubernetes:ConfigMap {
+    configMaps: [
+            {
+                mountPath: "/home/ballerina/data",
+                data: ["./conf/data.txt"],
+                defaultMode: 777
+            }
+        ]
```
Comment (hemikak) [inline kubernetes-extension-test/src/test/resources/default-mode/default_mode_config_map.bal]: Shall we fix this indentation ?

File: kubernetes-extension-annotations/src/main/ballerina/src/kubernetes/annotation.bal [hunk 7]
```diff
@@ -303,11 +303,13 @@ public const annotation PodAutoscalerConfig HPA on source service, source functi
 #
 # + mountPath - Mount path.
 # + readOnly - Is mount read only. Default is `true`.
+# + defaultMode - Default permission mode.
```
Comment (hemikak) [inline kubernetes-extension-annotations/src/main/ballerina/src/kubernetes/annotation.bal]: We will need to update the ReadMe.md file as well .

File: kubernetes-extension-test/src/test/java/org/ballerinax/kubernetes/test/DefaultModeTest.java [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+public class DefaultModeTest {
+
+    private static final Path BAL_DIRECTORY = Paths.get("src", "test", "resources", "default-mode");
+    private static final Path DOCKER_TARGET_PATH = BAL_DIRECTORY.resolve(DOCKER);
+    private static final Path KUBERNETES_TARGET_PATH = BAL_DIRECTORY.resolve(KUBERNETES);
+    private static final String DOCKER_IMAGE_SECRET = "default_mode_secret:latest";
+    private static final String DOCKER_IMAGE_CONFIG = "default_mode_config_map:latest";
+
+    /**
+     * Build bal file with deployment having annotations.
```
Comment (hemikak) [inline kubernetes-extension-test/src/test/java/org/ballerinax/kubernetes/test/DefaultModeTest.java]: Need to update the comment.

File: kubernetes-extension-test/src/test/java/org/ballerinax/kubernetes/test/DefaultModeTest.java [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+        Assert.assertEquals(volume.getSecret().getSecretName(), "helloworld-secret");
+        Assert.assertEquals(volume.getSecret().getDefaultMode().intValue(), 755);
+
+        KubernetesUtils.deleteDirectory(KUBERNETES_TARGET_PATH);
+        KubernetesUtils.deleteDirectory(DOCKER_TARGET_PATH);
+        KubernetesTestUtils.deleteDockerImage(DOCKER_IMAGE_SECRET);
+    }
+
+    @Test
+    public void defaultModeConfigTest() throws IOException, InterruptedException, KubernetesPluginException,
```
Comment (hemikak) [inline kubernetes-extension-test/src/test/java/org/ballerinax/kubernetes/test/DefaultModeTest.java]: Shall we add a comment to this ?

**Comment by anuruddhal** [comment]
@hemikak Fixed the issues.

**Pull request merged**